File: 0680_Valid Palindrome II/ValidPalindromeII.cs
public class Solution {
    public bool ValidPalindrome(string s) {
        if(String.IsNullOrEmpty(s)) return true;
        int start = 0, end = s.Length - 1;
        while(start < end){
            if(s[start] != s[end]){
                return IsPalindrome(s, start+1,end) || IsPalindrome(s, start, end-1);
            }else{
                start++;
                end--;
            }
        }

        return true;
    }

    private bool IsPalindrome(string s, int start, int end){
        while(start < end){
            if(s[start++]!=s[end--]) return false;
        }

        return true;
    }
}

File: 0680_Valid Palindrome II/ValidPalindromeII.java
class Solution {
    public boolean validPalindrome(String s) {
        int l = 0;
        int r = s.length() - 1;
        
        while(l < r) {
            if(s.charAt(l) != s.charAt(r)) {
                return isPalindrome(s, l+1, r) || isPalindrome(s, l , r-1);
            }
            l++;
            r--;
        }
        
        return true;
    }
    
    private boolean isPalindrome(String s, int l , int r) {
        while(l < r) {
            if(s.charAt(l) != s.charAt(r)) return false;
            l++;
            r--;
        }
        
        return true;
    }
}